backend/src/openbeheer/utils/gherkin_e2e.py

```diff
@@ -65,16 +76,97 @@ class Given(GherkinScenario):
         These steps are used to describe the initial situation before an action is taken.
         """
 
-        pass
+        def user_exists(self) -> User:
+            """
+            Creates a user for testing.
+            """
+
+            return UserFactory.create(username="johndoe", password="secret")
+
+        def objecttypen_service_exists(self) -> Service:
+            """
+            Creates a (fake) objecttypen service for testing and registers it in
+            API_CONFIG.
+            """
+
+            fake_service = ServiceFactory.create(
+                api_type=APITypes.orc,
+                api_root="http://localhost:8003/besluiten/api/v1",
+            )
+            api_config = APIConfig.get_solo()
+            api_config.objecttypen_api_service = fake_service
+            api_config.save()
+
+            return fake_service
+
+        def selectielijst_service_exists(self) -> Service:
+            """
+            Creates a selectielijst service for testing and registers it in API_CONFIG.
+            """
+
+            selectielijst_service = ServiceFactory.create(
+                api_type=APITypes.orc,
+                api_root="https://selectielijst.openzaak.nl/api/v1/",
+            )
+            api_config = APIConfig.get_solo()
+            api_config.selectielijst_api_service = selectielijst_service
+            api_config.save()
```

Comment:
We should use the `APIConfigFactory`, which helps with test isolation by clearing the previously created solo models and clears the cache.